src/Models/AZ3166Device.ts

```diff
@@ -150,8 +157,82 @@ export class AZ3166Device extends ArduinoDeviceBase {
     return true;
   }
 
-  async configDeviceSettings(): Promise<boolean> {
-    const configSelectionItems = this.getConfigDeviceSettingsMainOptions();
+  async configDeviceSettings(): Promise<void> {
+    // Select device settings type
+    const deviceSettingType = await this.selectDeviceSettingType();
+
+    let credentials = '';
+    let deviceSettingsType = '';  // For log info
+    switch (deviceSettingType) {
+      case ConfigDeviceOptions.CRC:
+        await this.generateCrc(this.channel);
+        return;
+      case ConfigDeviceOptions.ConnectionString:
+        // Get device connection string
+        credentials = await this.getDeviceConnectionString();
+
+        deviceSettingsType = 'device connection string';
+        await this.logAndSetCredentials(
+            credentials, ConfigDeviceOptions.ConnectionString,
+            deviceSettingsType);
+        return;
+      case ConfigDeviceOptions.DPS:
+        // Get DPS Credential
+        credentials = await this.getDPSCredentialsFromInput();
+
+        deviceSettingsType = 'UPS credentials';
+        await this.logAndSetCredentials(
+            credentials, ConfigDeviceOptions.ConnectionString,
+            deviceSettingsType);
+        return;
+      case ConfigDeviceOptions.UDS:
+        // Get UDS string
+        credentials = await this.getUDSStringFromInput();
+
+        deviceSettingsType = 'Unique Device String (UDS)';
+        await this.logAndSetCredentials(
+            credentials, ConfigDeviceOptions.ConnectionString,
+            deviceSettingsType);
+        return;
+      default:
+        throw new Error(`Internal Error: Unsupported device setting type: ${
+            deviceSettingType}.`);
+    }
+  }
+
+  // Private functions for configure device settings
+
+  /**
+   * Print credentials in log. Set credentials to device.
+   * Pop up information message suggesting configuration operation is
+   * successful.
+   * @param credentials device credentials
+   * @param deviceSettingType device setting type
+   * @param deviceSettingsType device settings type info to be print in warning
+   * window
+   */
+  private async logAndSetCredentials(
+      credentials: string, deviceSettingType: ConfigDeviceOptions,
+      deviceSettingsType: string): Promise<void> {
+    // Log Credentials
+    console.log(credentials);
+
+    // Set credentials
+    const res = await this.setDeviceConfig(
+        credentials, deviceSettingType);  // TODO: Mind the return value.
+    if (!res) {
+      throw new Error('Fail to flush configuration to device');
+    }
+
+    vscode.window.showInformationMessage(
+        `Successfully configure ${deviceSettingsType}.`);
+  }
+
+  /**
+   * Select device setting type.
+   */
+  private async selectDeviceSettingType(): Promise<string|undefined> {

```

Comment:
>Promise<string|undefined [](start = 43, length = 24)

Why not using the 'ConfigDeviceOptions' enum

Reply:
Fixed.